components/indexPage.tsx

```diff
@@ -0,0 +1,290 @@
+// Copyright 2021 the Deno authors. All rights reserved. MIT license.
+/** @jsx h */
+/** @jsxFrag Fragment */
+
+import {
+  apply,
+  css,
+  type DocNode,
+  type DocNodeKind,
+  Fragment,
+  h,
+  tw,
+} from "../deps.ts";
+import { type IndexStructure } from "../docs.ts";
+import { store, type StoreState } from "../shared.ts";
+import { type Child, take } from "../util.ts";
+import { MarkdownSummary } from "./markdown.tsx";
+import { IndexMeta } from "./meta.tsx";
+import { SideBarHeader } from "./sidebar.tsx";
+import { gtw } from "./styles.ts";
+
+const panel = css({
+  "& > input:checked ~ table": apply`hidden`,
+  "& > input:checked ~ label > img": apply`rotate-0`,
+});
+
+function getDocSummary(docNode: DocNode) {
+  if (docNode.jsDoc?.doc) {
+    const [summary] = docNode.jsDoc.doc.split("\n\n");
+    return summary;
+  }
+}
+
+function getModuleSummary(
+  mod: string,
+  entries: Map<string, DocNode[]>,
+): string | undefined {
+  const docNodes = entries.get(mod);
+  if (docNodes) {
+    for (const docNode of docNodes) {
+      if (docNode.kind === "moduleDoc") {
+        return getDocSummary(docNode);
+      }
+    }
+  }
+}
+
+function ExportSymbol(
+  { children, base, name, kind }: {
+    children: Child<string | undefined>;
+    base: string;
+    name: string;
+    kind: DocNodeKind;
+  },
+) {
+  const summary = take(children);
+  const href = `${base}/~/${name}`;
+  let linkClass;
+  switch (kind) {
+    case "class":
+      linkClass = tw`text-green(800 dark:400) font-bold hover:underline`;
+      break;
+    case "enum":
+      linkClass = tw`text-green(700 dark:500) font-bold hover:underline`;
+      break;
+    case "function":
+      linkClass = tw`text-cyan(800 dark:400) font-bold hover:underline`;
+      break;
+    case "interface":
+      linkClass = tw`text-cyan(900 dark:300) font-bold hover:underline`;
+      break;
+    case "typeAlias":
+      linkClass = tw`text-yellow(700 dark:500) font-bold hover:underline`;
+      break;
+    case "variable":
+      linkClass = tw`text-blue(700 dark:500) font-bold hover:underline`;
+      break;
+    case "namespace":
+      linkClass = tw`text-yellow(800 dark:400) font-bold hover:underline`;
+      break;
+    default:
+      linkClass = tw`hover:underline`;
+  }
+  return (
+    <tr>
+      <td class={tw`px-2 align-top`}>
+        <a href={href} class={linkClass}>{name}</a>
+      </td>
+      <td class={tw`px-2 align-top`}>
+        <MarkdownSummary>{summary}</MarkdownSummary>
+      </td>
```

Comment:
Small suggestion: increase vertical padding/margin between the rows. Right now they are very dense which makes it a bit hard to scan which summary corresponds to which symbol